### ISSUE TITLE
Thermos 20 -> 21 Exception in server tick loop

### ISSUE THREAD

**Issue by TsumiHokiro** [TCPR/Thermos#90]
I've just changed from .20 to .21 and server wouldn't start due to not being able to initialize net.minecraftforge.common.ChestGenHooks.

https://gist.github.com/TsumiHokiro/d0578e57d69c505fb35a

Edit: Derped version numbers.

**Comment by Robotia** [comment]
Talked about this with @TsumiHokiro believe this is an issue with either his setup or his hosting provider. 
If anyone else encounters this issue, please comment here.

**Comment by TsumiHokiro** [comment]
Alright, so I've been working on this and I got some more info.
As I thought, the crash report only reports one of the several errors that were found in this crash. Looking at the stats of the log I ascertained that more errors were found and got the following:
Railcraft,OpenBlocks, MagicBees, Calculator, Botania, Enchantmentplus, Randomthings, rftools, BiomesoPlenty, : Could not initialize class net.minecraftforge.common.ChestGenHooks
ArsMagica2: java.lang.VerifyError: Bad type on operand stack
Chisel: java.lang.NullPointerException: Cannot add variation in group RCAbyssalBlock for null block.
So as said in the Discord, only with Railcraft it surely works. I'm now trying to remove the less popular mods of this list, in special Ars Magica and see what happens. I do know that if I remove all of these I get past this phase and error in the next phase (bunch of UCHE mods).
So @Robotia, it does not seems to be wrong Jar, does not seems to be a host problem. Seems more like a mod conflict with Thermos but I'll continue to test and see what I can find.

**Comment by TsumiHokiro** [comment]
@Robotia Alright, I found out the problem and I'm closing it as it's beyond the power of Thermos to fix it, or rather, it's been decided not to mess with it and I don't blame anyone other than ReikaKalseki for his policy of not helping Cauldron and myself for being chained in parts to his content.
The mod causing the problem is the infamous DragonAPI who is trying to do something that breaks Thermos. Although I'm using the KC version of it, some recent change to the code of Thermos probably broke its compatibility.
This was the mod that was problematic that I mentioned in Discord but I failed to remove the DragonAPI, only removed Rotary, Reactor and Electricraft and left the most problematic one (How stupid!).
I'm not mentioning him here because he has already explicitly said that he won't support KCauldron (and probably, I assume, Thermos).
Since I'm also in the wrong (I'm using KC version of DragonAPI) I do not want to raise any concerns with this any further.
TL;DR: Stay away from DragonAPI and do yourself some good and avoid headaches :)

**Comment by Robotia** [comment]
I'm very sorry about this, if I wasn't worried that Reika might come after me for making it compatible with Thermos, I would help you out. :(

**Comment by TsumiHokiro** [comment]
It's ok. I completely understand how Reika doesn't help with making it easier for you. In fact, had I known before hand that Reika would say "I tried to communicate with the Dev about issues and he didn't respond me" and known for sure that he didn't try to contact Prototik  about KCauldron, I'd have avoided putting it on my server, even though his content IS GREAT. No matter how great his content may be, if he is not willing to support it, even if just to suggest what might be wrong, a minority, it's not worth the trouble.
I'll see if I can it working with the latest KCauldron version. If it is not, I'll try reaching Prototik about it. Worse come to worst, I'll just freeze meanwhile until I get to the point I can re-create the world and get rid of a headache.

**Comment by bookerthegeek** [comment]
Which is really really odd as every time we talk Reika is a very reasonable person. He just do sent suffer idiots. Ever.

[Rant]
That being said, I highly doubt that whatever the issue that exactly happened is due to an error in dragon api. I've only ever seen an issue with it that was brought to the surface due to Reika's mods refusing to overlook other people's mistakes. As for fixs to his code... have you seen his code? It beautifully. 
[/Rant]

Want me to go ask him? We get along pretty well.

**Comment by TsumiHokiro** [comment]
@bookerthegeek In case you haven't seen it in KCauldron's gitlab, I can find it for you the post where Reika is reported an error when a server was running KCauldron, Reika's outbreak saying that he has tried to reach the devs of it, interestingly enough one of the dev is the one who reported it to Reika, says they're not responsive nor tried to help him, he not so long after pushes his 6th rule that "People can't use their own/3rd party modified jars" in order to have his permission to run a modpack.
Interestingly enough, on the same day, Prototik publishes his own DragonAPI version for KC that fixed the error...
Who was wrong? Who was right? What I did see, however, is the reactor of Reika to the problem saying he did ask for help but received none, prohibits people from using his PUBLIC Source (Note, its not open. It's all rights reserved to him) in order to fix problems which his Code has when running in a modded environment (Modded server, not that he is not running in an already modded environment...) which he does not support and prohibits others from adapting his Code in order to be able to do it.
Now in his total defence, and also as a coward that I am, I DID NOT try to report to him the error, so I did not give him the opportunity to react differently to me. So maybe I'm mistaken, he can decide that it might be worthy to look over the case of KCauldron and Thermos, and in fact, I'd love that was the case. So if you feel that you might have an understanding with him, please, p-l-e-a-s-e, reach him and bring light to us. All this mess about his mods not working with KCauldron just brings headaches to the USERS of his mods, not so much to the server owners. Server owners can just decide "nope, too much trouble to put this into my world". Since I'm both, and I'm an avid user of his Mods, I find it so very sad the situation that is presented to me, in particular, through the happenings of others.
TL;DR: I'd love to if you decide to ask him. In fact, I wouldn't even mind if by this contact with him I could get the chance to see things change (for the better or for the worse) cause, seriously, I hate the state of things right now.

**Comment by TsumiHokiro** [comment]
Also, as you've mentioned before, his code is great. Even I who don't know much of coding respect his coding. It is not the issue here if he does bad coding or good coding (which he does).
The problem is how he decides to interact with Minecraft, not through Forge, which breaks how other mods, when running KCauldron/Thermos, expect to interact with Forge/Minecraft.
That is the issue. He makes his mods work, that's great. He breaks other mods (KCauldron/Thermos). Not so great. I just wish it would be possible to have his mods work alongside other mods.

**Comment by bookerthegeek** [comment]
Okily Dokily. I'll go ask. No promises.

**Comment by TsumiHokiro** [comment]
@bookerthegeek Here is the link which I got the info about the DragonAPI patch and what issued the whole drama: https://gitlab.prok.pw/KCauldron/KCauldron/issues/306

**Comment by ReikaKalseki** [comment]
As mentioned [here](https://github.com/ReikaKalseki/Reika_Mods_Issues/issues/523), this particular issue is already fixed. Also on that thread is some important context, including that I was for months only told "it crashes, nothing you can do" and that the KCauldron team was entirely unwilling to even explain the actual issue.

As for custom builds, I do not, and never will allow them because I cannot verify their content. I have had numerous instances in the past where people have tried slipping bugs, unauthorized changes, and *even malware* into the distributed jars. Given that the average end-user has absolutely no way of knowing that the problems arose due to modification and are not native to the mod, it has caused serious problems that I have had to clean up after.

**Comment by bookerthegeek** [comment]
:)

**Comment by TsumiHokiro** [comment]
@ReikaKalseki Although this issue has been fixed before, I guess Thermos has done something that has resurfaced this issue as this has happened just recently while using the new version from Thermos (or it was new then).
Also, I do understand your reasons for choosing to do so, and I completely respect them. I also do understand that hearing it from third-hand is terrible but given that there is "documented" (because the internet [github, gitlab, etc] documents them) material about the offense I did feel rather cautious about reporting to either side (I admit that I did not dig through the countless issues on the ReikaIssues).
@Robotia So now that Reika has come shed some light into the issue, do you think there is anything that could be done about the compatibility of his mods with Thermos or is it a still "no-go" stance that will be adopted?

**Comment by ReikaKalseki** [comment]
> @ReikaKalseki Although this issue has been fixed before, I guess Thermos has done something that has resurfaced this issue as this has happened just recently while using the new version from Thermos (or it was new then).

v11 has not yet been released.

**Comment by TomeWyrm** [comment]
v11 has been released now. Did that fix this error?

**Comment by ReikaKalseki** [comment]
I am still getting bug reports for Cauldron, so I rather doubt it. As seen [here](https://github.com/ReikaKalseki/Reika_Mods_Issues/issues/658#issuecomment-189830763), I suspect that the fix I was given is nonfunctional.

**Comment by TomeWyrm** [comment]
Well, that one doesn't include a crash log. I've been a contributor on your github long enough to expect the PEBCAK error until proven otherwise. Thus asking here for confirmation from people that don't make me worry for the collective intelligence of the human race.

**Comment by ProNoob135** [comment]
So, is there anything I can do about this? a different server jar? I still need plugins.

**Comment by TomeWyrm** [comment]
@ProNoob135 crash log please?

Odds are you're going to have to bug the Thermos folks because 19/20 times it's on their end, but seeing the log will let us know for sure.

**Comment by ProNoob135** [comment]
[crash-2016-10-08_17.04.05-server.txt](https://github.com/CyberdyneCC/Thermos/files/518007/crash-2016-10-08_17.04.05-server.txt)

**Comment by ProNoob135** [comment]
Nothing different with Thermos-1.7.10-1614-server.jar (latest Pre-Release)

**Comment by Robotia** [comment]
If you are able to edit server arguments, the quickest fix is adding -noverify. I am not sure about the cause of the underlying error.

**Comment by ReikaKalseki** [comment]
That is just going to cause a crash.

**Comment by ProNoob135** [comment]
How do I edit JVM arguments anyways(Thermos)? been wondering.(I'm new to this lol)

**Comment by TomeWyrm** [comment]
If it's anything like all the other server wrappers, you edit the portion of the file used to start the server that points at the .jar (usually named "start"). On windows this will be a .bat or .cmd file... though you might not know that because windows hides file extensions.

**Comment by ProNoob135** [comment]
There is no .bat file, I tried to make one but it didn't work, so I'm just launching the .jar. I always turn off the hidden file extensions :)

**Comment by ProNoob135** [comment]
Seems to have fixed it, thanks Reika. Also, it kills me and puts me into survival if I join in creative, is this a bug or is there a way to disable it?

**Comment by ReikaKalseki** [comment]
...

**Comment by ProNoob135** [comment]
Well now I'm getting a crash for no apparent reason. I won't waste anymore of your time though.

**Comment by Robotia** [comment]
@ReikaKalseki how does `-noverify` cause a crash? I have not experienced that before :confused:

**Comment by ReikaKalseki** [comment]
Because that simply ignores an invalid class. As soon as you need it, you will probably crash.

**Comment by Robotia** [comment]
The class is not invalid. A similar problem is encountered when stackmap frames are incorrectly rebuilt in Thermos when compiled on JDK8 source 8 target 8. Adding --noverify simply means Java will not verify stackmap frames, thus avoiding the problem.
